lib/MongoDB/_Protocol.pm

```diff
@@ -420,9 +689,30 @@ sub parse_reply {
         }
         }
         if ( length($msg) < $len )
-        || ( $opcode != OP_REPLY )
+        || ( ( $opcode != OP_REPLY ) && ( $opcode != OP_MSG ) )
         || ( $response_to != $request_id );
 
+
+    if ( $opcode == OP_MSG ) {
+        MongoDB::InternalError->throw('OP_MSG requires codec for parse_reply') unless defined $codec;
+        # TODO Extract and check checksum
+        my @sections = split_sections( $codec, substr $msg, P_MSG_PREFIX_LENGTH );
+        # We have none of the other stuff? maybe flags... and an array of docs? erm
+        return {
+          flags => {
+            checksum_present => vec( $bitflags, MSG_FB_CHECKSUM, 1 ),
+            query_failure    => vec( $bitflags, MSG_FB_MORE_TO_COME, 1 ),
+          },
+          docs => $sections[0]->{documents}->[0]
```

Comment:
Add a "XXX" comment that this assumes the server never sends a type 1 payload (which it currently doesn't).  

When the server adds that capability (as it may in 4.2), we'll need to combine the type 1 payload into the type 0 payload. (I.e. the opposite of prepare).  We don't need to do that now (since we have no tests for it), but I want a marker in the code to remind us for later.